server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/pageobjects/WebhooksPage.java

```diff
@@ -37,12 +43,98 @@ public WebhooksPage hasWebhook(String text) {
     return this;
   }
 
+  public WebhooksPage hasNoWebhooks() {
+    $(".boxed-group").shouldHave(text("No webhook defined"));
+    return this;
+  }
+
+  public WebhooksPage hasLatestDelivery(String webhookName) {
+    getWebhooks().find(text(webhookName)).should(exist).shouldNotHave(text("Never"));
+    return this;
+  }
+
   public WebhooksPage countWebhooks(Integer number) {
     getWebhooks().shouldHaveSize(number);
     return this;
   }
 
+  public WebhooksPage createWebhook(String name, String url) {
+    $(".js-webhook-create").shouldBe(visible).shouldBe(enabled).shouldNotHave(cssClass("disabled")).click();
+    modalShouldBeOpen("Create Webhook");
+    $("#webhook-name").shouldBe(visible).sendKeys(name);
+    $("#webhook-url").shouldBe(visible).sendKeys(url);
+    $("button[type='submit']").shouldBe(visible).click();
+    modalShouldBeClosed();
+    return this;
+  }
+
+  public  WebhooksPage createIsDisabled() {
+    $(".js-webhook-create").shouldBe(visible).shouldHave(cssClass("disabled")).click();
+    modalShouldBeClosed();
+    return this;
+  }
+
+  public WebhooksPage deleteWebhook(String webhookName) {
+    SelenideElement webhook = getWebhooks().find(text(webhookName)).should(exist);
+    webhook.$(".dropdown-toggle").shouldBe(visible).click();
+    webhook.$(".js-webhook-delete").shouldBe(visible).click();
+    modalShouldBeOpen("Delete Webhook");
+    $("button.button-red").shouldBe(visible).click();
+    modalShouldBeClosed();
+    return this;
+  }
+
+  public DeliveriesForm showDeliveries(String webhookName) {
+    SelenideElement webhook = getWebhooks().find(text(webhookName)).should(exist);
+    webhook.$(".dropdown-toggle").shouldBe(visible).click();
+    webhook.$(".js-webhook-deliveries").shouldBe(visible).click();
+    modalShouldBeOpen("Recent deliveries for " + webhookName);
+    return new DeliveriesForm($(".modal-body"));
+  }
+
+  public class DeliveriesForm {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make this a "static" inner class. ([squid:S2694](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10347_deliveries&issues=AWGUijmFQeDOAQ7I7lgA&open=AWGUijmFQeDOAQ7I7lgA)

[ISSUE_KEY]: AWGUijmFQeDOAQ7I7lgA